directives/animate/animate.js

```diff
@@ -0,0 +1,273 @@
+import { directive, PropertyPart } from 'lit-html';
+import { getComposedActiveElement, getNextFocusable } from '../../helpers/focus.js';
+import { getComposedParent, isComposedAncestor } from '../../helpers/dom.js';
+
+const stateMap = new WeakMap();
+const reduceMotion = matchMedia('(prefers-reduced-motion: reduce)').matches;
+const showTransitionDuration = 300;
+const hideTransitionDuration = 200;
+const moveYValue = 20;
+
+class AnimationState {
+
+	constructor(propertyPart) {
+
+		if (!(propertyPart instanceof PropertyPart) || propertyPart.committer.name !== 'animate') {
+			throw new Error('animation directives must be used with "animate" property');
+		}
+
+		this.id = 0;
+		this.clone = null;
+		this.elem = propertyPart.committer.element;
+		this.state = 'unknown';
+
+	}
+
+	async animate(info) {
+
+		if (this.clone === null) {
+			this.clone = document.createElement('div');
+		}
+		this.elem.parentNode.insertBefore(this.clone, this.elem);
+
+		const id = ++this.id;
+
+		let outerResolve;
+		const onTransitionEnd = () => {
+			this.clone.removeEventListener('transitionend', onTransitionEnd);
+			if (this.id === id) {
+				this.clone.remove();
+				this.clone = null;
+				if (info.elem.styleAttr !== null) {
+					this.elem.setAttribute('style', info.elem.styleAttr);
+				} else {
+					this.elem.removeAttribute('style');
+				}
+				info.onTransitionEnd();
+				this.dispatchEvent();
+				outerResolve();
+			}
+		};
+		this.clone.addEventListener('transitionend', onTransitionEnd);
+
+		Object.assign(this.clone.style, info.clone.start);
+		Object.assign(this.elem.style, info.elem.start);
+
+		if (info.elem.transition) {
+			await new Promise((r) => requestAnimationFrame(r));
+			Object.assign(this.elem.style, info.elem.transition);
+		}
+
+		await new Promise((r) => requestAnimationFrame(r));
+		Object.assign(this.clone.style, info.clone.end);
+		Object.assign(this.elem.style, info.elem.end);
+
+		return new Promise((resolve) => outerResolve = resolve);
+
+	}
+
+	dispatchEvent(timeout = 0) {
+		// need a timeout as when motion is reduced
+		// event fires too quickly to listen for
+		// it right after render
```

Comment:
It appears as though the unit tests for Edge (Chromium) at least in the BrowserStack environment are running with `prefers-reduced-motion` turned on. In that scenario, the event is firing before the tests can even wire up an event listener. So adding a short timeout that we'll only use in the reduced motion case.